src/extensions/agent_modules/cloud_communication/adps2_mqtt_client/src/adps_gen2.c

```diff
@@ -153,14 +154,14 @@ bool ReadAzureDPS2MqttSettings(AZURE_DPS_2_MQTT_SETTINGS* settings)
     if (!ADUC_AgentInfo_ConnectionData_GetBooleanField(
             agent_info, "dps.cleanSession", &settings->mqttSettings.cleanSession))
     {
-        Log_Info("CleanSession: %d", DEFAULT_ADPS_CLEAN_SESSION);
+        Log_Info("default CleanSession: %d", DEFAULT_ADPS_CLEAN_SESSION);
         settings->mqttSettings.cleanSession = DEFAULT_ADPS_CLEAN_SESSION;
     }
 
     if (!ADUC_AgentInfo_ConnectionData_GetUnsignedIntegerField(
```

Comment:
Looks like there **is** a GetUnsigned.  Places above where tmp<0 is being checked should instead use GetUnsigned, no?

Reply:
Tracking this for later. It seems logical but the DPS is working well and don't have time right now to investigate if the dps connection data can return a negative integer for that.